### DISCUSSION TITLE
How does the IF activity in Elsa 3.1.3 designer define a condition that evaluates to true or false?

### DISCUSSION THREAD

**Discussion by shxylgithub** [elsa-workflows/elsa-core#5704] [General]
How does the IF activity in Elsa 3.1.3 designer define a condition that evaluates to true or false?

**Comment by glime-ai** [comment]
The `If` activity in Elsa 3.1.3 designer allows you to define a condition that evaluates to true or false by setting the `Condition` property. This property expects an expression that, when evaluated, results in a boolean value (`true` or `false`). Based on the result of this condition, the workflow will proceed with the activities defined in the `Then` branch if the condition is `true`, or the `Else` branch if the condition is `false`.

In the designer interface, you may not see explicit `Then` and `Else` properties to set activities. Instead, you will see two buttons labeled `Then` and `Else` in the activity display, which represent sub-flowcharts. When you click on one of these buttons, you are taken to a sub-flowchart where you can drag and drop activities to define the logic for that branch of the `If` activity.

To summarize, you define the condition in the `Condition` attribute of the `If` activity, and then you can implement the logic for what happens when the condition is `true` or `false` by using the `Then` and `Else` sub-flowcharts.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
